cdk/ecs/lib/stacks/ecsStack.ts

```diff
@@ -0,0 +1,1118 @@
+        readonly dimension: string;
+        readonly match: string;
+    }>;
+    readonly replacements: Array<{
+        readonly target_dimension: string;
+        readonly value: string;
+    }>;
+    readonly action: string;
+};
+
+export class EcsClusterStack extends Stack {
+    public readonly cluster: Cluster;
+    private readonly securityGroups: SecurityGroup[];
+    private readonly ecsTaskRole: Role;
+    private readonly ecsTaskExecutionRole: Role;
+    private readonly subnets: ISubnet[];
+    private readonly ecrImagePrefix: string;
+    private readonly serviceDiscoveryStack: ServiceDiscoveryStack;
+    private readonly logStack: LogStack;
+    private readonly CLUSTER_NAME = 'ecs-pet-clinic-demo';
+    private readonly CONFIG_SERVER = 'pet-clinic-config-server';
+    private readonly DISCOVERY_SERVER = 'pet-clinic-discovery-server';
+    private readonly ADMIN_SERVER = 'pet-clinic-admin-server';
+    private readonly API_GATEWAY = 'pet-clinic-api-gateway';
+    private readonly VISITS_SERVICE = 'pet-clinic-visits-service';
+    private readonly CUSTOMERS_SERVICE = 'pet-clinic-customers-service';
+    private readonly VETS_SERVICE = 'pet-clinic-vets-service';
+    private DISCOVERY_SERVER_CW_CONFIG: MetricTransformationConfig;
+    private CONFIG_SERVER_CW_CONFIG: MetricTransformationConfig;
+    private VISIT_SERVICE_CW_CONFIG: MetricTransformationConfig;
+    private VET_SERVICE_CW_CONFIG: MetricTransformationConfig;
+    private CUSTOMERS_SERVICE_CW_CONFIG: MetricTransformationConfig;
+
+    constructor(scope: Construct, id: string, props: EcsClusterStackProps) {
+        super(scope, id, props);
+
+        this.cluster = new Cluster(this, 'EcsCluster', {
+            vpc: props.vpc,
+            clusterName: this.CLUSTER_NAME,
+        });
+
+        this.ecrImagePrefix = `${this.account}.dkr.ecr.${this.region}.amazonaws.com`; // retrive ECR image from the private repository
+
+        this.securityGroups = props.securityGroups;
+        this.ecsTaskRole = props.ecsTaskRole;
+        this.ecsTaskExecutionRole = props.ecsTaskExecutionRole;
+        this.subnets = props.subnets;
+        this.serviceDiscoveryStack = props.serviceDiscoveryStack;
+        this.logStack = props.logStack;
+        this.replaceRemoteServicesNames();
+
+        new CfnOutput(this, 'EcsClusterArn', { value: this.cluster.clusterArn });
+    }
+
+    createService(props: CreateServiceProps) {
+        // 1. create service discovery service
+        const DNSService = this.serviceDiscoveryStack.createService(props.serviceName);
+
+        // 2, create ECS service
+        const ecsService = new FargateService(this, `${props.serviceName}-ecs-service`, {
+            serviceName: props.serviceName,
+            taskDefinition: props.taskDefinition,
+            cluster: this.cluster,
+            securityGroups: this.securityGroups,
+            vpcSubnets: {
+                subnets: this.subnets,
+            },
+            assignPublicIp: true,
```

Comment:
Do we need to assign public ip to the ECS service? I don't remember where we use public ip to access the sample app.